friends/src/App.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import './App.css';
+import axios from 'axios';
+import UpdateForm from './Components/UpdateForm'
+import ListOfFriends from './Components/ListOfFriends'
+import FriendForm from './Components/FriendForm'
+import { Route, NavLink, } from "react-router-dom";
+
+class App extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      friends: [],
+      activeItem: []
+    };
+  }
+
+
+  componentDidMount() {
+    console.log('CDM now running');
+    axios
+      .get('http://localhost:5000/friends')
+      .then(res => {
+        this.setState({ friends: res.data });
+      })
+      .catch(err => {
+        console.log(err);
+        this.setState({ error: err });
+      });
+  }
+
+  addFriend = (e, friend) => {
+    e.preventDefault();
+    axios
+      .post('http://localhost:5000/friends', friend)
+      .then (res => {
+        this.setState({
+          friends: res.data
+        });
+        this.props.history.push('/')
+      })
+      .catch(err => {
+        console.log(err)
+      })
+  }
+
+  deleteFriend = (e, friend) => {
+
+    e.preventDefault();
+    console.log('I didnt like them anyways');
+    console.log(friend.id)
+    axios
+      .delete(`http://localhost:5000/friends/${friend.id}`)
+      .then(res => {
+        console.log(res.data);
+        this.setState({
+          friends: res.data
+        });
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  };
+
+  setUpdateForm = (e, item) => {
+    console.log("inside setting update form")
+    e.preventDefault();
+    this.setState({
+      activeItem: item
+    },
+    () => {
+      this.props.history.push('/update-form')
+    })
+  }
+
+  updateFriend = item => {
+    console.log("inside update friend")
+    axios 
+      .put(`http://localhost:5000/friends/${item.id}`, item)
+      .then (res => {
+        this.setState({
+          friends: res.data
+        });
+        this.props.history.push('/')
+      })
+      .catch(err => {
+        console.log(err)
+      })
+  }
+
+  render() {
+    return (
+      <div>
+        <nav>
```

Comment:
Small recommendation - I would make a separate component for the Nav and store this there.